test/runnable/class_destructors.d

```diff
@@ -0,0 +1,178 @@
+/**
+https://issues.dlang.org/show_bug.cgi?id=21799
+
+PERMUTE_ARGS:
+TRANSFORM_OUTPUT: remove_lines( Deprecation: The `delete` keyword has been deprecated)
+**/
+
+int main()
+{
+	testDeleteD();
+	testDeleteDScope();
+	testDeleteWithoutD();
+	testDeleteCpp();
+	testDeleteCppScope();
+	testDeleteWithoutCpp();
+	testThrowingDtor();
+	return 0;
+}
+
+enum forceCtfe = main();
+
+/*************************************************/
+
+class ParentD
+{
+	char[]* ptr;
+
+	~this()
+	{
+		*ptr ~= 'A';
+	}
+}
+
+class ChildD : ParentD
+{
+	~this()
+	{
+		*ptr ~= 'B';
+	}
+}
+
+void testDeleteD()
+{
+	char[] res;
+	ChildD cd = new ChildD();
+	cd.ptr = &res;
+	delete cd;
```

Comment:
Shouldn't you use `destroy(cd)` instead ?

Reply:
CTFE fails to execute `destroy` for `extern(D)` classes because it calls into an opaque druntime hook.
But that should be fixed in the future.